app/views/play/level/tome/editor/autocomplete.coffee

```diff
@@ -314,6 +314,54 @@ module.exports = class Autocomplete
     haveFindNearest = false
     autocompleteReplacement = level.get("autocompleteReplacement") ? []
     usedAutocompleteReplacement = []
+
+    fixlanguageSnippets = (doc, lang) ->
+      usedAutocompleteReplacement = []
+
+      if lang in ['java', 'cpp'] and not doc?.snippets?[lang] and doc?.snippets?.javascript
+        doc.snippets[lang] = doc.snippets.javascript
+
+      if lang in ['lua', 'coffeescript', 'python'] and not doc?.snippets?[lang] and (doc?.snippets?.python or doc?.snippets?.javascript)
+          # These are mostly the same, so use the Python or JavaScript ones if language-specific ones aren't available
+        doc.snippets[lang] = doc?.snippets?.python or doc.snippets.javascript
```

Comment:
I know this is how it used to be, but I wonder if we should update this to use our new `translateJS` function while we're at it.

Reply:
oh i just checked with translateJS. it not works same. so i won't use that right now. (some snippets is not translated but just rewrite for different langs. i.e. `moveRightLoopSnippet` to `while (true) {\n    hero.moveRight();\n    ${1:}\n}`) maybe we could add new translate function in aether-utils later.